packages/tracing/src/browser/request.ts

```diff
@@ -216,7 +216,9 @@ export function xhrCallback(
   const xhr = handlerData.xhr.__sentry_xhr__;
 
   // check first if the request has finished and is tracked by an existing span which should now end
-  if (handlerData.endTimestamp && handlerData.xhr.__sentry_xhr_span_id__) {
+  if (handlerData.endTimestamp) {
```

Comment:
we probably want this logic to be like:

```ts
if (handlerData.endTimestamp) {
  const spanId = handlerData.xhr.__sentry_xhr_span_id__;
  if (spanId) {
    const span = spans[spanId];
    if (span) {
      span.setHttpStatus(xhr.status_code);
      span.finish();

      // eslint-disable-next-line @typescript-eslint/no-dynamic-delete
      delete spans[spanId];
    }
  }
  return;
}
```

This is also more bundle size efficient

Reply:
hi, thanks for looking into this.  

I have refactored into using the `spanId` variable as suggested.  
But I'm using eager-return style because it modifies less codes and I also believe multiple nesting levels is less readable. Not sure if it's significant to the bundle size?